src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -223,4 +243,74 @@ private static void loadMaxUnits(final Unit transport, final List<Unit> canBeTra
     }
   }
 
+  private static Map<Unit, List<Unit>> findTransportsThatUnitsCouldUnloadFrom(final Collection<Unit> units,
+      final Collection<Unit> transports) {
+    final List<Unit> canBeTransported = Match.getMatches(units, Matches.UnitCanBeTransported);
+    final List<Unit> canTransport = Match.getMatches(transports, Matches.UnitCanTransport);
+    final Map<Unit, List<Unit>> result = new LinkedHashMap<>();
+    for (final Unit unit : canBeTransported) {
+      final List<Unit> transportOptions = new ArrayList<Unit>();
+      for (final Unit transport : canTransport) {
+        if (containsEquivalentUnit(unit, TransportTracker.transporting(transport))) {
+          transportOptions.add(transport);
+        }
+      }
+      result.put(unit, transportOptions);
+    }
+    return result;
+  }
+
+  private static Map<Unit, List<Unit>> sortByTransportOptionsAscending(
+      final Map<Unit, List<Unit>> unitToPotentialTransports) {
+    final Map<Unit, List<Unit>> result = new LinkedHashMap<>();
+    unitToPotentialTransports.entrySet().stream()
```

Comment:
One of the best Java 8 stream usages I done. Really makes sorting LinkedHashMaps much cleaner than before.